Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MATLABAnalytics/MATLABAnalytic.tsx

```diff
@@ -0,0 +1,107 @@
+
+
+
+import * as React from 'react';
+import MATLABAnalyticInfo from './MATLABAnalyticInfo';
+import MATLABAnalyticSQLSettings from './MATLABAnalyticSQLSetting';
+import { useAppSelector, useAppDispatch } from '../hooks';
+import { MATLABAnalyticSlice, MATLABAnalyticEventTypeSlice, MATLABAnalyticAssetTypeSlice } from '../Store/Store';
+import { TabSelector, Warning } from '@gpa-gemstone/react-interactive';
+
+declare var homePath: string;
+
+export default function MATLABAnalytic(props: { AnalyticID: number }) {
+    const dispatch = useAppDispatch();
+
+    const record = useAppSelector((state) => MATLABAnalyticSlice.Datum(state, props.AnalyticID));
+    const eventTypeRecords = useAppSelector(MATLABAnalyticEventTypeSlice.Data);
+    const assetTypeRecords = useAppSelector(MATLABAnalyticAssetTypeSlice.Data);
+
+    const status = useAppSelector(MATLABAnalyticSlice.Status);
+    const eventTypeStatus = useAppSelector(MATLABAnalyticEventTypeSlice.Status);
+    const assetTypeStatus = useAppSelector(MATLABAnalyticAssetTypeSlice.Status);
+
+    const [tab, setTab] = React.useState<'info' | 'settings'>('info');
+    const [showRemove, setShowRemove] = React.useState<boolean>(false);
+
+    React.useEffect(() => {
+        if (status == 'unintiated' || status == 'changed')
+            dispatch(MATLABAnalyticSlice.Fetch());
+    }, [status]);
+
+    React.useEffect(() => {
+        if (eventTypeStatus == 'unintiated' || eventTypeStatus == 'changed')
+            dispatch(MATLABAnalyticEventTypeSlice.Fetch(props.AnalyticID));
+    }, [eventTypeStatus]);
+
+    React.useEffect(() => {
+        if (assetTypeStatus == 'unintiated' || assetTypeStatus == 'changed')
+            dispatch(MATLABAnalyticAssetTypeSlice.Fetch(props.AnalyticID));
+    }, [assetTypeStatus]);
+
+    React.useEffect(() => {
+        sessionStorage.setItem('MATLABAnalytic.Tab', JSON.stringify(tab));
+    }, [tab]);
+
+    function Delete() {
+        dispatch(MATLABAnalyticSlice.DBAction({ verb: 'DELETE', record }));
+        eventTypeRecords.forEach((item) => dispatch(MATLABAnalyticEventTypeSlice.DBAction({ verb: 'DELETE', record: item })));
+        assetTypeRecords.forEach((item) => dispatch(MATLABAnalyticAssetTypeSlice.DBAction({ verb: 'DELETE', record: item })));
+
+        window.location.href = homePath + 'index.cshtml?name=MATLABAnalytics';
+    }
+
+    if (record == null) return null;
+    return (
+        <div style={{ width: '100%', height: window.innerHeight - 63, maxHeight: window.innerHeight - 63, overflow: 'hidden', padding: 15 }}>
+            <div className="row">
+                <div className="col">
+                    <h2>MATLAB Analytic</h2>
+                </div>
+                <div className="col">
+                    <button className="btn btn-danger pull-right" hidden={record == null}
+                        onClick={() => setShowRemove(true)}>Delete Analytic (Permanent)</button>
```

Comment:
```suggestion
                        onClick={() => setShowRemove(true)}>Delete Analytic</button>
```
I think adding (permanent) here is unnecessary since we have the warning pop-up. If you feel differently, feel free to ignore this comment.